### PR TITLE
Use versionName for version for Android app projects

### LINKED ISSUE #79: Android App Versions not dedected
Running SPDX sbooms on an android project results in the following warning

```
spdx sboms require a version but project: app has no specified version
```

I guess this is again an android thing, the version is part of the gradle file and can be found in:

```
android {
    defaultConfig {
        applicationId "XYZ"
        minSdk 29
        targetSdk 34
        versionCode 1
        versionName "1.0"
```

_Originally posted by @PeterHasse in https://github.com/spdx/spdx-gradle-plugin/issues/49#issuecomment-1845251229_

### PR DESCRIPTION
**Pull request by eygraber**
Fixes #79 

Technically the correct way to do this would be through the Android variants API. However that could get complicated with this library's Target API, so I didn't do that for now.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -22,6 +29,7 @@ dependencies {
     implementation("org.apache.maven:maven-model-builder:3.9.6")
     implementation("org.apache.maven:maven-model:3.9.6")
     implementation("com.google.guava:guava:32.1.3-jre")
+    implementation("com.android.tools.build:gradle:8.2.0")
```
Comment (loosebazooka) [inline build.gradle.kts:32]: can this be provided? And we consume it from the classpath of the android project itself? I'm not sure I want this loaded into the classpath for non-android sbom projects.

File: src/main/java/org/spdx/sbom/gradle/project/ProjectInfo.java [hunk 2]
```diff
@@ -49,6 +50,20 @@ static ProjectInfo from(Project project) {
         .build();
   }
 
+  static String version(Project project) {
```
Comment (loosebazooka) [inline src/main/java/org/spdx/sbom/gradle/project/ProjectInfo.java:53]: need to update `ProjectInfo.from` to use this, but I would prefer to re-arrange this so AppExtension (line 54) is calculated in the if(android) block. That might help us avoid explicitly shipping with the android build dependency?

File: src/functionalTest/java/org/spdx/SpdxSbomPluginFunctionalTest.java [hunk 3]
```diff
@@ -253,12 +253,24 @@ public void canResolveVariantDependencies() throws IOException {
...
+            + "  }\n"
+            + "}");
+
     writeString(
         appKotlinBuildFile,
         "plugins {\n"
             + " kotlin(\"android\") version \"1.9.21\"\n"
             + "  id(\"org.spdx.sbom\")\n"
-            + "  id(\"com.android.application\") version \"8.1.4\"\n"
+            + "  id(\"com.android.application\")\n"
```
Comment (loosebazooka) [inline src/functionalTest/java/org/spdx/SpdxSbomPluginFunctionalTest.java:273]: we could probably use this test to check if version is read.

File: src/main/java/org/spdx/sbom/gradle/project/ProjectInfo.java (hunk 2 above)
Comment (eygraber) [reply src/main/java/org/spdx/sbom/gradle/project/ProjectInfo.java:53]: Making the android plugin a `compileOnly` dependency leads to a `ClassNotFoundException` even if we move the `AppExtension` code into the block that checks for the presence of the android plugin.

**Comment by eygraber** [comment]
I ended up going with reflection because of classpath issues with accessing AGP from inside the build service, so there's no need to depend on AGP

**Comment by eygraber** [comment]
Thomas Broyer on the Gradle slack pointed out that if `withPluginClasspath` is removed from the test then it will work without reflection. I tested this and it is true, but it would require publishing the plugin to local or something like that. Another option would be to keep the test using `version` instead of `versionName`, release 1.0.0, and then update the test to use the actual plugin and use `versionName`.

I'm ambivalent because the reflection is pretty simple in this case.

**Comment by loosebazooka** [comment]
Ok sgtm

**Review by loosebazooka — APPROVED** [review]

**Pull request merged**